### PR TITLE
Feature/tidy test

### PR DESCRIPTION
**Pull request by starcolon**
## What changes were proposed in this pull request?

Tidy up the redundant Spark life cycle setup and tear down of some tests.

## How was this patch tested?

The patch doesn't change any of the logics. So the existing unit tests should not break.

## Related links or issues (optional)

n/a

### CODE REVIEW

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.scalatest.{BeforeAndAfter, FlatSpec, Matchers}
+
+trait SparkTestLifeCycle extends FlatSpec with BeforeAndAfter {
```
Comment (zhangxiaoli73) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala]: SparkContextLifeCycle

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  var sc: SparkContext = null
+  def nodeNumber: Int
+  def coreNumber: Int
+  def appName: String
```
Comment (zhangxiaoli73) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala]: set a default  name

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  before {
+    Engine.init(nodeNumber, coreNumber, true)
+    val conf = new SparkConf().setMaster("local[1]").setAppName(appName)
```
Comment (zhangxiaoli73) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/SparkTestLifeCycle.scala]: should be local[$coreNumber], not local[1]

**Comment by zhangxiaoli73** [comment]
Hi ,thank you for your pr. Maybe you can do some changes:
1. Change "SparkTestLifeCycle" to "SparkContextLifeCycle" and add some comments about how to use it
2. Add default values for variables, as users may not assign values by themselves
3. Maybe your trait should just replace "BeforeAndAfter" in other unit tests.

**Comment by starcolon** [comment]
Hi, I agree with your suggestions. Some of other compatible tests which also make use of [[sc]] now also use [[SparkContextLifeCycle]]. I leave some incompatible tests which [[SparkContextLifeCycle]] cannot apply as they are.

**Comment by zhangxiaoli73** [comment]
ok. Please merge your nine commits into one.

**Comment by starcolon** [comment]
okay, squashed :)

**Comment by zhangxiaoli73** [comment]
PR has code style errors. Please use "mvn clean verify -DskipTests" to check code style.

**Comment by starcolon** [comment]
fixed. just noticed `mvn test....` wont trigger the scala style checker. it should build find now

**Pull request merged**